rustls/src/error.rs

```diff
@@ -94,6 +94,15 @@ pub enum Error {
     /// or too large.
     BadMaxFragmentSize,
 
+    /// This [`rustls::crypto::signer::SigningKey`] has not implemented [`rustls::crypto::signer::SigningKey::public_key()`].
+    UnimplementedSubjectPublicKeyInfo,
+
+    /// We failed to parse DER into an end-entity certificate.
+    FailedToParseEndEntityCert(webpki::Error),
```

Comment:
We avoid having `webpki` types in our public interface. Please reuse `Error::InvalidCertificate` here.

Reply:
Gotcha. I replaced the code that necessitated this error in the first place with `ParsedCertificate` in [38dd19e](https://github.com/rustls/rustls/pull/1954/commits/38dd19e0ea9471274a9cf67c06b0aad494966a49)